src/async-wrap.cc

```diff
@@ -544,9 +544,19 @@ void AsyncWrap::Initialize(Local<Object> target,
       isolate,
       uid_fields_ptr,
       uid_fields_count * sizeof(*uid_fields_ptr));
+  Local<Float64Array> farray = Float64Array::New(uid_fields_ab, 0,
+                                                 uid_fields_count);
   FORCE_SET_TARGET_FIELD(target,
                          "async_uid_fields",
-                         Float64Array::New(uid_fields_ab, 0, uid_fields_count));
+                         farray);
+
+#if ENABLE_TTD_NODE
+  if (s_doTTRecord || s_doTTReplay) {
+    unsigned int refcount = 0;
+    JsAddRef(*farray, &refcount);
```

Comment:
We could change this to a "JsTTDNotifyLongLivedReferenceAdd(*farray)" to avoid keeping the array alive but this array is global to the module so it shouldn't make a difference (but I do like changing it since that is the better API use).